BLD: build fails with

### Bug report

**Bug summary**

We've had a couple instabilities in CI lately, so I'm only catching this now after disantengling the rest of them, but it seems that builds have been failing with the following error for a couple days
```
yt/utilities/lib/platform_dep.h(28): error C2169: 'log2': intrinsic function, cannot be defined
```
I don't know what changed upstream, but it doesn't seem to be numpy or Cython.

--- a/tests/ci_install.sh
+++ b/tests/ci_install.sh
@@ -34,7 +34,7 @@ if [[ "${RUNNER_OS}" == "Windows" ]] && [[ ${dependencies} != "minimal" ]]; then
     # keep in sync: setup.cfg
     while read requirement; do conda install --yes $requirement; done < tests/windows_conda_requirements.txt
 else
-    python -m pip install --upgrade pip
+    python -m pip install --upgrade "pip != 22.1"
     python -m pip install --upgrade wheel
 
     # // band aid

--- a/yt/utilities/lib/geometry_utils.pyx
+++ b/yt/utilities/lib/geometry_utils.pyx
@@ -15,7 +15,7 @@ import numpy as np
 cimport cython
 cimport numpy as np
 from cython cimport floating
-from libc.math cimport copysign, fabs
+from libc.math cimport copysign, fabs, log2
 from libc.stdlib cimport free, malloc
 
 from yt.utilities.lib.fp_utils cimport fclip, i64clip
@@ -40,7 +40,6 @@ cdef extern from "math.h":
     double fabs(double x) nogil
 
 cdef extern from "platform_dep.h":
-    double log2(double x) nogil
     long int lrint(double x) nogil
 
 @cython.cdivision(True)

--- a/yt/utilities/lib/platform_dep.h
+++ b/yt/utilities/lib/platform_dep.h
@@ -25,9 +25,6 @@ static __inline double fmax(double x, double y){
 static __inline double fmin(double x, double y){
     return (x < y) ? x : y;
 }
-static __inline double log2(double x) {
-    return log(x) * M_LOG2E;
-}
 
 /* adapted from http://www.johndcook.com/blog/cpp_erf/
    code is under public domain license */

